src/main/java/XbeLoader/XbeImageHeader.java

```diff
@@ -22,37 +22,95 @@
 import ghidra.util.task.TaskMonitor;
 
 public class XbeImageHeader {
-	public long   magic;
-	public byte[] signature;
-	public long   baseAddr;
-	public long   headersSize;
-	public long   imageSize;
-	public long   imageHeaderSize;
-	public long   timestamp;
-	public long   certificateAddr;
-	public long   sectionCount;
-	public long   sectionHeadersAddr;
-	public long   initFlags;
-	public long   entryAddr;
-	public long   tlsAddr;
-	public long   peStackCommit;
-	public long   peHeapReserve;
-	public long   peHeapCommit;
-	public long   peBaseAddr;
-	public long   peImageSize;
-	public long   peChecksum;
-	public long   peTimestamp;
-	public long   debugPathnameAddr;
-	public long   debugFilenameAddr;
-	public long   debugUnicodeFilenameAddr;
-	public long   kernThunkAddr;
-	public long   importDirAddr;
-	public long   libVersionsCount;
-	public long   libVersionsAddr;
-	public long   kernLibVersionAddr;
-	public long   xapiLibVersionAddr;
-	public long   logoAddr;
-	public long   logoSize;
+	public enum XbeType {
+		RETAIL(0),
+		DEBUG(1),
+		CHIHIRO(2);
+
+		private int value;
+
+		private XbeType(int setValue) {
+			value = setValue;
+		}
+
+		public int getValue() {
+			return value;
+		}
+	}
+
+	private long[] xorEntryPoint = {
+		0xA8FC57ABL, // Retail
+		0x94859D4BL, // Debug
+		0x40B5C16EL  // Chihiro
+	};
+
+	private long[] xorKernelThunk = {
+		0x5B6D40B6L, // Retail
+		0xEFB1F152L, // Debug
+		0x2290059DL  // Chihiro
+	};
```

Comment:
nit:
1. Not a fan of disconnecting `xorEntryPoint` and `xorKernelThunk` in independent arrays.
2. Not a fan of creating a pseudo-index into these arrays as it might be incorrectly mistaken for an actual XBE feature (~"type 1 XBE" / "type 2 XBE" when referring to Debug or Chihiro... which is not a thing).

I'd rather have seen something like:

```
class ChihiroXbe {
  const long xorEntryPoint: ...
  const long xorKernelThunk: ...
};
class DebugXbe {
  const long xorEntryPoint: ...
  const long xorKernelThunk: ...
};
```
*(Or however one would do this in Java)*

And then directly remember which kind of type it is by pointing at these classes.
However, there might be even better ways in Java.

Reply:
I agree. However, I'm porting as-is since it's base on our codebase currently. Anyone can improve it after this PR.

On the side note, I have not been using java language for years. So, past and present codes are made from Notepad++ and verified code style format do match.